lib/matplotlib/collections.py

```diff
@@ -856,36 +851,82 @@ def get_linewidth(self):
     def get_linestyle(self):
         return self._linestyles
 
-    def update_scalarmappable(self):
-        """Update colors from the scalar mappable array, if it is not None."""
+    def _set_mappable_flags(self):
+        edge0 = self._edge_is_mapped
+        face0 = self._face_is_mapped
         if self._A is None:
+            self._edge_is_mapped = False
+            self._face_is_mapped = False
+            # return False  # Nothing to map
+        else:
+            # Typical mapping: centers, not edges.
+            self._face_is_mapped = True
+            self._edge_is_mapped = False
+
+            # Make the colors None or a string. (If None, it is a default.)
+            fc = self._original_facecolor
+            if not (fc is None or isinstance(fc, str)):
+                fc = 'array'
+            ec = self._original_edgecolor
+            if not (ec is None or isinstance(ec, str)):
```

Comment:
```suggestion
            if ec is not None and not isinstance(ec, str):
```
As above: Same but IMHO easier to read.

Reply:
This section got reworked so the line is no longer there.